With this change, `interface_gui.py` fails on macOS, if you build using depends:
```bash
/sq/z88fhjzj0b19ftsd2_bjrmjm0000gn/T/test_runner_₿_🏃_20260730_103518
Remaining jobs: [interface_gui.py]
1/1 - interface_gui.py failed, Duration: 3 s

stdout:
2026-07-30T09:35:18.801091Z TestFramework (INFO): PRNG seed is: 6738883396313921420
2026-07-30T09:35:18.856332Z TestFramework (INFO): Initializing test directory /var/folders/sq/z88fhjzj0b19ftsd2_bjrmjm0000gn/T/test_runner_₿_🏃_20260730_103518/interface_gui_0
2026-07-30T09:35:21.693745Z TestFramework (ERROR): Unexpected exception:
Traceback (most recent call last):
  File "/Users/bitcoin/test/functional/test_framework/test_framework.py", line 141, in main
    self.setup()
    ~~~~~~~~~~^^
  File "/Users/bitcoin/test/functional/test_framework/test_framework.py", line 272, in setup
    self.setup_network()
    ~~~~~~~~~~~~~~~~~~^^
  File "/Users/bitcoin/test/functional/test_framework/test_framework.py", line 359, in setup_network
    self.setup_nodes()
    ~~~~~~~~~~~~~~~~^^
  File "/Users/bitcoin/build/test/functional/interface_gui.py", line 30, in setup_nodes
    super().setup_nodes()
    ~~~~~~~~~~~~~~~~~~~^^
  File "/Users/bitcoin/test/functional/test_framework/test_framework.py", line 381, in setup_nodes
    self.start_nodes()
    ~~~~~~~~~~~~~~~~^^
  File "/Users/bitcoin/test/functional/test_framework/test_framework.py", line 513, in start_nodes
    node.wait_for_rpc_connection()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/bitcoin/test/functional/test_framework/test_node.py", line 353, in wait_for_rpc_connection
    raise FailedToStartError(self._node_msg(
        f'bitcoind exited with status {self.process.returncode} during initialization. {str_error}'))
test_framework.test_node.FailedToStartError: [node 0] bitcoind exited with status -11 during initialization. 
2026-07-30T09:35:21.750767Z TestFramework (INFO): Not stopping nodes as test failed. The dangling processes will be cleaned up later.
```

--- a/test/functional/interface_gui.py
+++ b/test/functional/interface_gui.py
@@ -21,11 +21,9 @@ class GuiTest(BitcoinTestFramework):
         self.skip_if_no_gui()
         # On Windows, bitcoin.exe exits immediately when launching bitcoin-gui.exe,
         # causing the test framework's process monitor to see a premature node exit.
-        # On macOS, bitcoin-qt's Cocoa code assumes NSApp is initialized, but the
-        # minimal Qt platform plugin skips that, causing crashes.
-        # Both issues are likely fixable.
-        if platform.system() in ("Windows", "Darwin"):
-            raise SkipTest("bitcoin-gui test not supported on Windows or macOS")
+        # This issue is likely fixable.
+        if platform.system() == "Windows":
+            raise SkipTest("bitcoin-gui test not supported on Windows")
 
     def setup_nodes(self):
         self.extra_init = [{"use_gui": True}]

--- a/test/functional/test_framework/test_node.py
+++ b/test/functional/test_framework/test_node.py
@@ -282,6 +282,11 @@ class TestNode():
         subp_env = dict(os.environ, LIBC_FATAL_STDERR_="1")
         if self.use_gui:
             subp_env.setdefault("QT_QPA_PLATFORM", "minimal")
+            if platform.system() == "Darwin":
+                # QMacStyle assumes a Cocoa platform window, which the minimal platform
+                # does not provide. In particular, painting a QGroupBox can make Qt call
+                # addSubview: on an invalid native object (QTBUG-49686).
+                subp_env.setdefault("QT_STYLE_OVERRIDE", "fusion")
             subp_env.setdefault("LC_ALL", "nl_NL.UTF-8") # Set language to try to trigger translation bugs
             if sys.platform.startswith("linux") and "XDG_RUNTIME_DIR" not in subp_env:
                 # Qt prints warnings to stderr when XDG_RUNTIME_DIR is unset or has wrong

